security/tpm/src/tpm.ts

```diff
@@ -288,8 +285,28 @@ export class TpmSecurityClient  {
     });
   }
 
+  private _readPersistentPrimary(name: string, handle: TPM_HANDLE, callback: (err: Error, resultPublicKey: TPMT_PUBLIC) => void): void {
+    this._tpm.allowErrors().ReadPublic(handle, (resp: tss.ReadPublicResponse) => {
+      let rc = this._tpm.getLastResponseCode();
+      debug('ReadPublic(' + name + ') returned ' + TPM_RC[rc] +  (rc === TPM_RC.SUCCESS ? '; PUB: ' + resp.outPublic.toString() : ''));
+      if (rc !== TPM_RC.SUCCESS) {
+        debug('readPersistentPrimary failed for: ' + name);
+        callback(new errors.SecurityDeviceError('Authorization unable to find a persistent identity key.  RC value: ' + TPM_RC[this._tpm.getLastResponseCode()].toString()), null);
+      } else {
+        callback(null, resp.outPublic);
+      }
+    });
+  }
+
+
   private _signData(dataToSign: Buffer, callback: (err: Error, signedData: Buffer) => void): void {
 
+    /*Codes_SRS_NODE_TPM_SECURITY_CLIENT_06_013: [If `signWithIdentity` is invoked without a previous successful invocation of `activateIdentityKey`, the callback will be invoked with `err` of `InvalidOperationError`] */
+    if (!this._idKeyPub) {
+      callback(new errors.InvalidOperationError('activateIdentityKey must be invoked before any signing is attempted.'), null);
+      return;
```

Comment:
nit: the `,null` is unnecessary and you could in theory just `return callback(new errors...)` that way it doesn't look like something happens after the callback (even if it's just a return)

Reply:
Well as it was written, the callback required signedData to be returned.  Added the ?